### PR TITLE
fix(ext/node): implement AES GCM cipher

### PR DESCRIPTION
**Pull request by littledivy**
Adds support for AES-GCM 128/256 bit keys in `node:crypto` and `setAAD()`, `setAuthTag()` and `getAuthTag()`

Uses https://github.com/littledivy/aead-gcm-stream

Fixes https://github.com/denoland/deno/issues/19836 https://github.com/denoland/deno/issues/20353

### CODE REVIEW

File: ext/node/ops/crypto/cipher.rs [hunk 1]
```diff
@@ -159,10 +214,37 @@ impl Decipher {
...
+  fn set_aad(&mut self, aad: &[u8]) {
+    use Decipher::*;
+    match self {
+      Aes128Gcm(decipher) => {
+        decipher.set_aad(aad);
+      }
+      Aes256Gcm(decipher) => {
+        decipher.set_aad(aad);
+      }
+      _ => {}
```
Comment (bartlomieju) [inline ext/node/ops/crypto/cipher.rs:244]: Shouldn't it be `todo!()` here?

File: ext/node/ops/crypto/cipher.rs [hunk 2]
```diff
@@ -103,10 +121,37 @@ impl Cipher {
...
+  fn set_aad(&mut self, aad: &[u8]) {
+    use Cipher::*;
+    match self {
+      Aes128Gcm(cipher) => {
+        cipher.set_aad(aad);
+      }
+      Aes256Gcm(cipher) => {
+        cipher.set_aad(aad);
+      }
+      _ => {}
```
Comment (bartlomieju) [inline ext/node/ops/crypto/cipher.rs:151]: Ditto

**Review by bartlomieju — APPROVED** [review]

File: ext/node/ops/crypto/cipher.rs (hunk 1 above)
Comment (littledivy) [reply ext/node/ops/crypto/cipher.rs:244]: AAD is only for authentication ciphers (GCM, CCM, OCB, and chacha20-poly1305) out of which we only support GCM.

**Comment by rawkakani** [comment]
hi @bartlomieju @littledivy, which date is the next release for this to be available?

**Comment by bartlomieju** [comment]
Around September 19th

**Pull request merged**